src/main/java/org/opentripplanner/graph_builder/GraphBuilder.java

```diff
@@ -172,6 +169,22 @@ public static GraphBuilder create(
         StreetLinkerModule streetLinkerModule = new StreetLinkerModule();
         streetLinkerModule.setAddExtraEdgesToAreas(config.areaVisibility);
         graphBuilder.addModule(streetLinkerModule);
+
+        if ( hasOsm ) {
+            if (OTPFeature.PruneIslands.isOn()) { // original pruning
+                PruneFloatingIslands pruneFloatingIslands = new PruneFloatingIslands();
+                pruneFloatingIslands.setPruningThresholdIslandWithoutStops(config.pruningThresholdIslandWithoutStops);
+                pruneFloatingIslands.setPruningThresholdIslandWithStops(config.pruningThresholdIslandWithStops);
+                graphBuilder.addModule(pruneFloatingIslands);
+            }
+            if (OTPFeature.PruneNoThru.isOn()) { // more advanced island pruning
+                PruneNoThruIslands pruneNoThruIslands = new PruneNoThruIslands(streetLinkerModule);
+                pruneNoThruIslands.setPruningThresholdIslandWithoutStops(config.pruningThresholdIslandWithoutStops);
+                pruneNoThruIslands.setPruningThresholdIslandWithStops(config.pruningThresholdIslandWithStops);
+                graphBuilder.addModule(pruneNoThruIslands);
+            }
+        }
```

Comment:
Could you add a comment here, I expect this to have a dependency to some of the above operations, since it is moved down after the gtfs/netx/osm section. What do this depend on?

Reply:
Good point, I added a clarifying comment. Furthermore, I now removed the 'old' pruning because it will corrupt the graph when executed after stop linking. I does not make sense to start fixing bugs of the old less accurate method, when a better solution is available.